### PR TITLE
New sample: Define clustering feature reduction

### PR DESCRIPTION
**Pull request by duffh**
# Description

Implemented new sample `Custom feature clustering`.

## Type of change

- New sample implementation

## Platforms tested on

- [x] WPF .NET 8
- [x] WPF Framework
- [x] WinUI
- [x] MAUI WinUI
- [x] MAUI Android
- [x] MAUI iOS
- [ ] MAUI MacCatalyst

## Checklist

- [x] Self-review of changes
- [x] All changes work as expected on all affected platforms
- [x] There are no warnings related to changes
- [x] Code is commented and follows .NET conventions and standards
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] No unrelated changes have been made to any other code or project files
- [x] Screenshots are correct size and display in description tab (800 x 600 on Windows, MAUI mobile platforms use the MAUI Windows screenshot)

### CODE REVIEW

File: src/Samples.Shared/Resources/FeaturedSamples.xml [hunk 1]
```diff
@@ -14,6 +14,7 @@
 			<Sample>AddCustomDynamicEntityDataSource</Sample>
 			<Sample>AddDynamicEntityLayer</Sample>
 			<Sample>AddVectorTiledLayerFromCustomStyle</Sample>
+      <Sample>CustomFeatureClustering</Sample>
```
Comment (williambohrmann3) [inline src/Samples.Shared/Resources/FeaturedSamples.xml:17]: nitpick: line indention

File: src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/CustomFeatureClustering.xaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
+﻿<ContentPage x:Class="ArcGIS.Samples.CustomFeatureClustering.CustomFeatureClustering"
+             xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:esriTK="clr-namespace:Esri.ArcGISRuntime.Toolkit.Maui;assembly=Esri.ArcGISRuntime.Toolkit.Maui"
+             xmlns:esriUI="clr-namespace:Esri.ArcGISRuntime.Maui;assembly=Esri.ArcGISRuntime.Maui">
+    <Grid ColumnDefinitions="*, Auto" RowDefinitions="Auto, *">
+        <esriUI:MapView x:Name="MyMapView" Style="{DynamicResource EsriSampleGeoView}" />
+        <Border Style="{DynamicResource EsriSampleControlPanel}">
```
Comment (williambohrmann3) [inline src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/CustomFeatureClustering.xaml:8]: Control panel is rendering 90% off screen for me on iPhone XS.

**Review by williambohrmann3 — COMMENTED** [review]
Nice, this is a great addition to our samples offering. Left a few minor nitpicks.

File: src/Samples.Shared/Resources/FeaturedSamples.xml (hunk 1 above)
Comment (duffh) [reply src/Samples.Shared/Resources/FeaturedSamples.xml:17]: Good catch

File: src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/readme.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Custom feature clustering
+
+Add client side custom feature reduction to a web map that does not have an existing defined feature reduction.
```
[the remark this answers is not in the document]
Comment (duffh) [reply src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/readme.md:3]: It was tricky to word initially, I'm essentially trying to say that this sample demonstrates how to add custom feature reduction in the application when it is not already defined on a web map. 

So in this case "an existing defined feature reduction" is a feature reduction which exists and has been defined with some parameters.

File: src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/readme.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+![Image of custom feature clustering](customfeatureclustering.jpg)
+
+## Use case
+
+Feature clustering can be used to dynamically aggregate groups of points that are within proximity of each other in order to represent each group with a single symbol. Such grouping allows you to see patterns in the data that are difficult to visualize when a layer contains hundreds or thousands of points that overlap and cover each other. Custom feature clustering allows users to add feature clustering to layers without predefined feature reduction. This is useful when the layer does not have feature reduction enabled by default or when the default feature reduction needs to be overridden.
+
+## How to use the sample
+
+Tap the draw clusters button to enable clustering on the feature layer. Interact with the clustering properties to change the cluster radius, max scale and to enable cluster labels. Tap the map to see the cluster feature count and aggregate fields in the popup.
```
[the remark this answers is not in the document]
Comment (duffh) [reply src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/readme.md:13]: I prefer that, it's more concise.

File: src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/CustomFeatureClustering.xaml (hunk 2 above)
Comment (duffh) [reply src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/CustomFeatureClustering.xaml:8]: I'll run this on iOS shortly to see if I can find a solution.

Comment (williambohrmann3) [reply src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/CustomFeatureClustering.xaml:8]: I should've added - this is an issue when rotating device orientation to landscape.

File: src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/readme.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/readme.md:3]: understood, thanks for clarifying

**Review by williambohrmann3 — APPROVED** [review]
thanks for addressing my nitpicks, LGTM.

File: src/MAUI/readme.md [hunk 5]
```diff
@@ -95,6 +95,7 @@
 * [Control annotation sublayer visibility](Maui.Samples/Samples/Layers/ControlAnnotationSublayerVisibility) - Use annotation sublayers to gain finer control of annotation layer subtypes.
 * [Create and save KML file](Maui.Samples/Samples/Layers/CreateAndSaveKmlFile) - Construct a KML document and save it as a KMZ file.
 * [Create feature collection layer (Portal item)](Maui.Samples/Samples/Layers/FeatureCollectionLayerFromPortal) - Create a feature collection layer from a portal item.
+* [Custom feature clustering](Maui.Samples/Samples/Layers/CustomFeatureClustering) - Add client side custom feature reduction to a web map that does not have an existing defined feature reduction.
```
Comment (pMaske) [inline src/MAUI/readme.md:98]:
```suggestion
* [Custom feature clustering](Maui.Samples/Samples/Layers/CustomFeatureClustering) - Add client side feature reduction on a point feature layer that is not pre-configured with clustering.
```

File: src/WinUI/readme.md [hunk 6]
```diff
@@ -95,6 +95,7 @@
 * [Control annotation sublayer visibility](ArcGIS.WinUI.Viewer/Samples/Layers/ControlAnnotationSublayerVisibility) - Use annotation sublayers to gain finer control of annotation layer subtypes.
 * [Create and save KML file](ArcGIS.WinUI.Viewer/Samples/Layers/CreateAndSaveKmlFile) - Construct a KML document and save it as a KMZ file.
 * [Create feature collection layer (Portal item)](ArcGIS.WinUI.Viewer/Samples/Layers/FeatureCollectionLayerFromPortal) - Create a feature collection layer from a portal item.
+* [Custom feature clustering](ArcGIS.WinUI.Viewer/Samples/Layers/CustomFeatureClustering) - Add client side custom feature reduction to a web map that does not have an existing defined feature reduction.
```
Comment (pMaske) [inline src/WinUI/readme.md:98]: Please update to match update comment above.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/CustomFeatureClustering/readme.metadata.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{
+    "category": "Layers",
+    "description": "Add client side custom feature reduction to a web map that does not have an existing defined feature reduction.",
```
Comment (pMaske) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/CustomFeatureClustering/readme.metadata.json:3]: Please update to match feedback above.

File: src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/readme.metadata.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
+{
+    "category": "Layers",
+    "description": "Add client side custom feature reduction to a web map that does not have an existing defined feature reduction.",
```
Comment (pMaske) [inline src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/readme.metadata.json:3]:
```suggestion
    "description": "Add client side feature reduction on a point feature layer that is not pre-configured with clustering.",
```

File: src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/CustomFeatureClustering.xaml (hunk 2 above)
Comment (duffh) [reply src/MAUI/Maui.Samples/Samples/Layers/CustomFeatureClustering/CustomFeatureClustering.xaml]: I've tested it. As it works and displays ok in portrait I think it's ok for now as I don't think the fix is a small piece of effort.

File: src/MAUI/Maui.Samples/Samples/Layers/DefineClusteringFeatureReduction/DefineClusteringFeatureReduction.xaml [hunk 9]
```diff
@@ -0,0 +1,87 @@
+﻿<ContentPage x:Class="ArcGIS.Samples.DefineClusteringFeatureReduction.DefineClusteringFeatureReduction"
+             xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:esriTK="clr-namespace:Esri.ArcGISRuntime.Toolkit.Maui;assembly=Esri.ArcGISRuntime.Toolkit.Maui"
+             xmlns:esriUI="clr-namespace:Esri.ArcGISRuntime.Maui;assembly=Esri.ArcGISRuntime.Maui">
+    <Grid ColumnDefinitions="*, Auto" RowDefinitions="Auto, *">
```
Comment (williambohrmann3) [inline src/MAUI/Maui.Samples/Samples/Layers/DefineClusteringFeatureReduction/DefineClusteringFeatureReduction.xaml:6]: Can you use the Dynamic Grid resource for this? That will fix the landscape orientation issue once #1317 is completed.

File: src/MAUI/Maui.Samples/Samples/Layers/DefineClusteringFeatureReduction/readme.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
+# Define clustering feature reduction
+
+Add client side feature reduction on a point feature layer that is not pre-configured with clustering.
+
+![Image of define clustering feature reduction](defineclusteringfeaturereduction.jpg)
+
+## Use case
+
+Feature clustering can be used to dynamically aggregate groups of points that are within proximity of each other in order to represent each group with a single symbol. Such grouping allows you to see patterns in the data that are difficult to visualize when a layer contains hundreds or thousands of points that overlap and cover each other. Users can add feature clustering to point feature layers without predefined feature reduction. This is useful when the layer does not have feature reduction enabled by default or when the existing feature reduction properties need to be overridden.
```
Comment (pMaske) [inline src/MAUI/Maui.Samples/Samples/Layers/DefineClusteringFeatureReduction/readme.md:9]:
```suggestion
Feature clustering can be used to dynamically aggregate groups of points that are within proximity of each other in order to represent each group with a single symbol. Such grouping allows you to see patterns in the data that are difficult to visualize when a layer contains hundreds or thousands of points that overlap and cover each other. Users can add feature clustering to point feature layers. This is useful when the layer does not have the feature reduction defined or when the existing feature reduction properties need to be overridden.
```

File: src/WPF/WPF.Viewer/Samples/Layers/DefineClusteringFeatureReduction/readme.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
+# Define clustering feature reduction
+
+Add client side feature reduction on a point feature layer that is not pre-configured with clustering.
+
+![Image of define clustering feature reduction](DefineClusteringFeatureReduction.jpg)
+
+## Use case
+
+Feature clustering can be used to dynamically aggregate groups of points that are within proximity of each other in order to represent each group with a single symbol. Such grouping allows you to see patterns in the data that are difficult to visualize when a layer contains hundreds or thousands of points that overlap and cover each other. Users can add feature clustering to point feature layers without predefined feature reduction. This is useful when the layer does not have feature reduction enabled by default or when the existing feature reduction properties need to be overridden.
```
Comment (pMaske) [inline src/WPF/WPF.Viewer/Samples/Layers/DefineClusteringFeatureReduction/readme.md:9]:
```suggestion
Feature clustering can be used to dynamically aggregate groups of points that are within proximity of each other in order to represent each group with a single symbol. Such grouping allows you to see patterns in the data that are difficult to visualize when a layer contains hundreds or thousands of points that overlap and cover each other. Users can add feature clustering to point feature layers. This is useful when the layer does not have the feature reduction defined or when the existing feature reduction properties need to be overridden.
```

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/DefineClusteringFeatureReduction/readme.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
+# Define clustering feature reduction
+
+Add client side feature reduction on a point feature layer that is not pre-configured with clustering.
+
+![Image of define clustering feature reduction](DefineClusteringFeatureReduction.jpg)
+
+## Use case
+
+Feature clustering can be used to dynamically aggregate groups of points that are within proximity of each other in order to represent each group with a single symbol. Such grouping allows you to see patterns in the data that are difficult to visualize when a layer contains hundreds or thousands of points that overlap and cover each other. Users can add feature clustering to point feature layers without predefined feature reduction. This is useful when the layer does not have feature reduction enabled by default or when the existing feature reduction properties need to be overridden.
```
Comment (pMaske) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/DefineClusteringFeatureReduction/readme.md:9]:
```suggestion
Feature clustering can be used to dynamically aggregate groups of points that are within proximity of each other in order to represent each group with a single symbol. Such grouping allows you to see patterns in the data that are difficult to visualize when a layer contains hundreds or thousands of points that overlap and cover each other. Users can add feature clustering to point feature layers. This is useful when the layer does not have the feature reduction defined or when the existing feature reduction properties need to be overridden.
```

**Review by pMaske — APPROVED** [review]
One minor update in sample description, otherwise looks good

**Pull request merged**